hub/core/meta/encode/chunk_id.py

```diff
@@ -209,3 +209,15 @@ def __getitem__(
             else:
                 break
         return output
+
+    def _pop(self) -> List[ENCODING_DTYPE]:
+        chunk_ids_for_last_sampe = self[-1]
+        if len(chunk_ids_for_last_sampe) > 1:
+            self._encoded = self._encoded[: -len(chunk_ids_for_last_sampe)]
+            return chunk_ids_for_last_sampe
+        elif self._encoded[-1][LAST_SEEN_INDEX_COLUMN] == 1:
```

Comment:
I'm assuming this is for the case in which the last chunk had 1 sample and is removed, in that case it shouldn't be ==1 but rather we should look at the previous row and the current row's difference and check that to be equal to 1.